packages/core/src/parseAmplifyGen2.ts

```diff
@@ -0,0 +1,267 @@
+import { APIGraphQLConfig, GraphQLAuthMode } from './singleton/API/types';
+import {
+	CognitoUserPoolConfigMfaStatus,
+	OAuthProvider,
+} from './singleton/Auth/types';
+import {
+	AuthType,
+	Gen2AuthMFAConfiguration,
+	Gen2OAuthIdentityProviders,
+} from './singleton/gen2/types';
+import { Gen2Config, LegacyConfig, ResourcesConfig } from './singleton/types';
+
+export function isGen2Config(
+	config: ResourcesConfig | LegacyConfig | Gen2Config,
+): config is Gen2Config {
+	return (config as Gen2Config).version === '1';
+}
+
+export function parseGen2Config(gen2Config: Gen2Config): ResourcesConfig {
+	const config: ResourcesConfig = {};
+
+	if (gen2Config?.storage) {
+		/* eslint-disable camelcase */
+		const { bucket_name, aws_region } = gen2Config.storage;
+		config.Storage = {
+			S3: {
+				bucket: bucket_name,
+				region: aws_region,
+			},
+		};
+	}
+
+	if (gen2Config?.auth) {
+		const {
+			user_pool_id,
+			user_pool_client_id,
+			identity_pool_id,
+			password_policy,
+			mfa_configuration,
+			mfa_methods,
+			unauthenticated_identities_enabled,
+			oauth,
+			username_attributes,
+			standard_required_attributes,
+		} = gen2Config.auth;
+
+		if (identity_pool_id) {
+			config.Auth = {
+				Cognito: {
+					userPoolId: user_pool_id,
+					userPoolClientId: user_pool_client_id,
+					identityPoolId: identity_pool_id,
+				},
+			};
+		} else {
+			config.Auth = {
+				Cognito: {
+					userPoolId: user_pool_id,
+					userPoolClientId: user_pool_client_id,
+				},
+			};
+		}
```

Comment:
Seems like you can reduce some duplication since both conditions result in a `Cognito` object with `userPoolId` and `userPoolClientId`
```suggestion
		config.Auth = {
			Cognito: {
				userPoolId: user_pool_id,
				userPoolClientId: user_pool_client_id,
			},
		};
		if (identity_pool_id) {
			config.Auth.Cognito.identityPoolId = identity_pool_id,
		}
```

Reply:
There was an issue with the Union type but I reduced this as much as possible

Thanks